src/bun.js/api/bun/h2_frame_parser.zig

```diff
@@ -0,0 +1,2440 @@
+            FrameType.HTTP_FRAME_SETTINGS => this.handleSettingsFrame(header, bytes[FrameHeader.byteSize..]) + FrameHeader.byteSize,
+            FrameType.HTTP_FRAME_WINDOW_UPDATE => this.handleWindowUpdateFrame(header, bytes[FrameHeader.byteSize..], stream) + FrameHeader.byteSize,
+            FrameType.HTTP_FRAME_HEADERS => this.handleHeadersFrame(header, bytes[FrameHeader.byteSize..], stream) + FrameHeader.byteSize,
+            FrameType.HTTP_FRAME_DATA => this.handleDataFrame(header, bytes[FrameHeader.byteSize..], stream) + FrameHeader.byteSize,
+            FrameType.HTTP_FRAME_CONTINUATION => this.handleContinuationFrame(header, bytes[FrameHeader.byteSize..], stream) + FrameHeader.byteSize,
+            FrameType.HTTP_FRAME_PRIORITY => this.handlePriorityFrame(header, bytes[FrameHeader.byteSize..], stream) + FrameHeader.byteSize,
+            FrameType.HTTP_FRAME_PING => this.handlePingFrame(header, bytes[FrameHeader.byteSize..], stream) + FrameHeader.byteSize,
+            FrameType.HTTP_FRAME_GOAWAY => this.handleGoAwayFrame(header, bytes[FrameHeader.byteSize..], stream) + FrameHeader.byteSize,
+            FrameType.HTTP_FRAME_RST_STREAM => this.handleRSTStreamFrame(header, bytes[FrameHeader.byteSize..], stream) + FrameHeader.byteSize,
+            else => {
+                this.sendGoAway(header.streamIdentifier, ErrorCode.PROTOCOL_ERROR, "Unknown frame type", this.lastStreamID);
+                return bytes.len;
+            },
+        };
+    }
+
+    const DirectWriterStruct = struct {
+        writer: *H2FrameParser,
+        pub fn write(this: *const DirectWriterStruct, data: []const u8) !bool {
+            this.writer.write(data);
+            return true;
+        }
+    };
+
+    fn toWriter(this: *H2FrameParser) DirectWriterStruct {
+        return DirectWriterStruct{ .writer = this };
+    }
+    pub fn setEncoding(this: *H2FrameParser, globalObject: *JSC.JSGlobalObject, callframe: *JSC.CallFrame) callconv(.C) JSValue {
+        JSC.markBinding(@src());
+        const args_list = callframe.arguments(1);
+        if (args_list.len < 1) {
+            globalObject.throw("Expected encoding argument", .{});
+            return .zero;
+        }
+        this.handlers.binary_type = BinaryType.fromJSValue(globalObject, args_list.ptr[0]) orelse {
+            const err = JSC.toInvalidArguments("Expected 'binaryType' to be 'arraybuffer', 'uint8array', 'buffer'", .{}, globalObject).asObjectRef();
+            globalObject.throwValue(err);
+            return .zero;
+        };
+
+        return JSC.JSValue.jsUndefined();
+    }
+
+    pub fn loadSettingsFromJSValue(this: *H2FrameParser, globalObject: *JSC.JSGlobalObject, options: JSC.JSValue) bool {
+        if (options.isEmptyOrUndefinedOrNull() or !options.isObject()) {
+            globalObject.throw("Expected settings to be a object", .{});
+            return false;
+        }
+
+        if (options.get(globalObject, "headerTableSize")) |headerTableSize| {
```

Comment:
`if (options.getOptional(globalObject, i32, "headerTableSize"))` I think, this will handle errors for you